### ISSUE TITLE
Update to apollo-server 0.2

### ISSUE THREAD

**Issue by tomitrescak** [apollostack/meteor-integration#24]
Hi, this no longer works with Apollo server 2.0+

I managed to solve most of the issues, but I have no idea how to approach the filling of context with meteor users information, since server 2.0 now uses context as a simple object.

**Comment by helfer** [comment]
@tomitrescak Do you have an example of what you're trying to do? You can still pass anything you want as the context, I believe. The only case where it won't work is if you're also using connectors from graphql-tools, because those will require context to be an object.

**Comment by helfer** [comment]
Apollo Server 0.2 has a lot of changes though, so it's possible that the errors you're seeing are due to something else.

**Comment by tomitrescak** [comment]
I'm trying to do exactly the same thing as before, that I want to read authorisation headers and fill in the userId. Context is initialised on server start, so how can I do this per request?

**Comment by helfer** [comment]
Instead of passing an options object to the server, you can pass it a function that takes the request as input.

**Comment by tomitrescak** [comment]
I tried that and it did not work, the function never evaluated :/

**Comment by tomitrescak** [comment]
Oh, ... i was testing that with console.log ... but probably this log never appears in the console as it runs in express, am I right ?

**Comment by tomitrescak** [comment]
I'll try that again. Will you be updating this project to 2.0+ by any chance?

**Comment by lorensr** [comment]
I should get to it next week if no one else does

**Comment by paralin** [comment]
I've been using a really simplified version of this package with the Accounts crap pulled out of it and it seems to be working fine with 0.2x.

**Comment by tomitrescak** [comment]
I was getting the "..must run within fiber" method. Would you care to share your server config ?

**Comment by wuzhuzhu** [comment]
@tomitrescak I runs into the same problem of 'must run within fiber' error, have any idea for now?

**Comment by jangerhofer** [comment]
Took a stab at updating to v0.2 in #26.

**Comment by lorensr** [comment]
Hi all, please try out `meteor add apollo@0.1.0-beta_1` with v0.2 support and report on whether it works for you, thanks!

**Comment by rdickert** [comment]
@lorensr This works, but seems to require Meteor 1.4, so this might require a note in the readme. I tried it first with my app (Meteor 1.3.3.1), then with a clone of simple-todos-react (Meteor 1.3.2.4). Both throw a `Use of const in strict mode.` error as seen elsewhere. I couldn't upgrade my own app (which uses std:accounts-ui and std:accounts-basic like the Meteor starter kit), but I was able to get simple-todos-react to run without errors when running Meteor 1.4.0.1 (I stopped before doing anything testable in Apollo).

I did have to run `meteor npm i --save body-parser@^1.15.2 graphql-tools@^0.6.2` to satisfy dependencies.

**Comment by lorensr** [comment]
Oh right, thanks – I'll update the package.js

**Comment by hlehmann** [comment]
I'am still having the error 500 `Invalid options provided to ApolloServer: Can't wait without a fiber`

**Comment by lorensr** [comment]
@hlehmann thanks for trying it out – is your app open-source, or would you be able to make a small reproduction that I could take a look at?

**Comment by hlehmann** [comment]
I'am making a small repro but first of all I have lost access to http://localhost:3000/graphql

**Comment by lorensr** [comment]
Sorry, try `/graphiql` – the new version's documentation is here: https://github.com/apollostack/docs/pull/167/files

**Comment by hlehmann** [comment]
Thx it's better ! After few tests I find that the error occur when you are logged in.

**Comment by hlehmann** [comment]
It works well when I download the package inside the `packages` directory but not when it's from atmosphere.

**Comment by rdickert** [comment]
@lorensr `meteor add apollo@0.1.0-beta_2` seems to be working perfectly in my app so far 🎉 

I did have to abandon the std:accounts packages to get to Meteor 1.4. We were looking for an alternative and didn't find anything really viable, so @jamielob made a simple package that wraps the original accounts-ui using thereactivestack:blazetoreact – [accounts-ui-react](https://github.com/okgrow/accounts-ui-react) that will hopefully be robust and low-maintenance. Have a look (comments welcome!), and if you are interested, I can do a PR for the Meteor Starter Kit to replace the std packages once we put it on Atmosphere.

**Comment by lorensr** [comment]
Thanks Robert! @jamielob that was easier than fixing std? 😆 Guess we can switch to ui-react for starter kit then, thanks :)

**Comment by jamielob** [comment]
Heh, @rdickert and I toyed with the idea of forking std but there are 18 open issues on the base repo alone and we didn't want to take on something that big right now.  Besides, who will be able to resist the charms of Frankenstein-UI.

**Comment by rdickert** [comment]
@lorensr Perhaps I was a bit hasty in declaring victory...

I had only checked at the level of Graphiql, which works fine. However, I too have gotten the `Invalid options provided to ApolloServer: Can't wait without a fiber` error on the client. I have done a [repro using the Meteor Starter Kit](https://github.com/okgrow/meteor-starter-kit/tree/0.2-fiber-repro). In the repro I did not update apollo-client or react-apollo, but it's the same if you do. if you go to [http://localhost:3000/graphiql](http://localhost:3000/graphiql) you can see that Graphiql is still fine (I had to remove the user check as context is not passed).

**Comment by lorensr** [comment]
Thanks for the repro! Oh, 1.4.1, haha. I'll take a look whenever it finishes downloading 😊

**Comment by lorensr** [comment]
Okay, using @rdickert's repro and local modified `meteor/apollo`, if you take out the async/await from:

https://github.com/apollostack/meteor-integration/blob/master/main-server.js#L28-L46

then it has the same error. But if you then also return early before the `users.findOne`, it's fixed:

![image](https://cloud.githubusercontent.com/assets/251288/17820537/47e97d4c-661b-11e6-8504-7589d1fa59b2.png)

@stubailo @helfer are you familiar enough with Fiberland and what changed in either apollo-server v0.2 or meteor v1.4 to know the fix? I have no idea 😁 Here are the changes since the last known working version of this repo: https://github.com/apollostack/meteor-integration/compare/4c89f40e0f9f76ce229a3c1200d5a7739e03f9f5...HEAD

Also related error, with apollo-server 0.1: https://github.com/apollostack/meteor-integration/issues/21

**Comment by Siyfion** [comment]
What's the status of this now? Seems like we've fallen behind now and I'd _really_ like to see us supporting the latest version of `apollo-server` and `apollo-client`, especially as the main Apollo docs point people at this package!

**Comment by lorensr** [comment]
Sorry, you're right, it would be good. I didn't get past that error. I think @stubailo is going to take a look at it 👍

**Comment by rdickert** [comment]
@Siyfion I rechecked it, and things seem to be working now. My PR against the starter kit was just merged, so maybe [give it a spin](https://github.com/apollostack/meteor-starter-kit).

**Comment by Siyfion** [comment]
@rdickert I've copied the way it's all integrated in the starter-kit and it seems to be working now, you're right. However, I had tried it before with the server code in the way the docs state:

```js
import { createApolloServer } from 'meteor/apollo';
import schema from '/imports/api/schema';
import resolvers from '/imports/api/resolvers';
createApolloServer({
  graphiql: true,
  pretty: true,
  schema,
  resolvers,
});
```

And it didn't seem to work, now.. I may have missed something _else_ than this change, but it's the only intentional difference that I am aware of.

**Comment by rdickert** [comment]
@Siyfion The way to set it up has changed, and the docs don't reflect those changes yet. There is a   PR that revises the docs ([see it here](https://github.com/jangerhofer/docs/blob/06481cad8c56bf85ee805bfe595c7a20328f32e0/source/apollo-client/meteor.md)). @lorensr maybe it's time to merge that and perhaps publish meteor-integration 0.1.0 to atmosphere w/o the beta flag?

**Comment by Siyfion** [comment]
Ah I see... shame that I didn't spot that earlier!

@rdickert & @lorensr I think you're right, you need to get the doc changes pulled and the 0.1.0 released properly, as you are indeed correct that it works fine! Perhaps the docs could also be duplicated in the README.md for this project?

**Comment by lorensr** [comment]
Haha, I was seeing this:
1. `meteor-starter-kit` using atmosphere `apollo`: working
2. `meteor-starter-kit` using local `apollo` at same tag: same fiber error as above
Solution to 2 was to `cd packages/apollo && rm -rf node_modules`

**Comment by lorensr** [comment]
Released `apollo@0.1.0`, which supports `'apollo-server': '^0.2.1'`. Thanks everyone!

**Comment by rdickert** [comment]
> Solution to 2 was to `cd packages/apollo && rm -rf node_modules`

oh, wow
